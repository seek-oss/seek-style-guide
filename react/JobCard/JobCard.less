@import (reference) "~seek-asia-style-guide/theme";
@job-card-section-space: 0;
@job-card-label-indent: @grid-base;
@job-card-list-indent: @grid-gutter-width;
@job-card-text-space: 0;
@job-card-bottom-text-space: @grid-base;
@job-card-logo-indent: @grid-gutter-width;

.root {
  width: 100%;
  margin-bottom: 0;
  border-style: solid;
  border-bottom-width: 1px;
  border-color: @saGrey4;
  @media @desktop {
    padding: 0 @grid-base * 2;
  }
}

.headerSection {
  padding-bottom: @job-card-section-space;
}

.highlightedBg {
  background-color: @actionPastelBlue;
}

.company {
  padding-bottom: @grid-base * 4;
  color: @saGrey1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.greyLabel {
  padding: @job-card-text-space @job-card-label-indent;
  margin-right: @job-card-label-indent;
  background-color: @saGrey5;
  color: @saGrey2;
}

.classifiedLabel {
  &:extend(.greyLabel);
}

.confidentialLabel {
  &:extend(.greyLabel);
}

.featuredLabel {
  padding: @job-card-text-space @job-card-label-indent;
  margin-right: @job-card-label-indent;
  background-color: @panelPink;
  color: @saWhite;
}

.positionTitle {
  color: @actionBlue;
  padding-bottom: @job-card-text-space;
  display: inline;
  word-wrap: break-word;
}

.titleKeyword {
  color: darken(@actionBlue, 15%);
}

.sellingPointsList {
  .rawText();
  list-style: disc;
  padding-left: @job-card-list-indent;
  color: @saGrey3;
  margin-left: @grid-base;
}

.sellingPoint {
  padding-bottom: @job-card-text-space;
  color: @saGrey2;
}

.sellingPointsSection {
  padding-top: @job-card-section-space;
}

.jobDescriptionSection {
  display: none;
  @media @desktop {
    padding-top: @job-card-section-space;
    display: flex;
  }
}

.withDescription {
  padding-bottom: @grid-base * 2;
}

.bodyDescriptionText {
  color: @saGrey1;
  padding-bottom: @job-card-text-space;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
}

.bodySection {
  padding-top: @job-card-section-space;
  padding-bottom: @job-card-section-space;
  display: flex;
  justify-content: space-between;
}

.footerSection {
  display: flex;
  justify-content: space-between;
}

.footerLeft {
  flex-basis: auto;
  overflow: hidden;
}

.jobInfoContainer {
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
}

.jobInfoList {
  padding-bottom: (@grid-base * 4);
}

.jobInfo {
  padding-bottom: @job-card-text-space;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  color: @saGrey2;
  /* stylelint-disable-next-line */
  & span {
    vertical-align: middle;
  }
}

.jobInfoIcon {
  color: @saGrey4;
  margin-right: @grid-base;
}

.postingDuration {
  padding-bottom: @job-card-bottom-text-space;
  color: @saGrey2;
}

.companyPicWrapper {
  display: none;
  @media @desktop {
    width: @grid-base * 50;
    min-width: @grid-base * 50;
    height: @grid-base * 35;
    max-height: 100%;
    display: flex;
    justify-content: flex-end;
    padding-left: @grid-base * 2;
    flex: 1;
  }
}

.companyPic {
  max-width: 100%;
  max-height: 100%;
  display: flex;
  object-fit: contain;
  align-self: flex-start;
}

.companyLogoWrapper {
  width: @grid-base * 20;
  min-width: @grid-base * 20;
  height: @grid-base * 20;
  margin-left: @job-card-logo-indent;
  display: flex;
  align-items: center;
  align-self: flex-end;
}

.companyLogo {
  max-width: 100%;
  max-height: 100%;
  margin: 0 auto;
  display: flex;
  align-self: flex-start;
}

.highlight {
  font-weight: @sa-bold;
  color: @saBlack;
}

.companyName {
  font-weight: @sa-semi-bold;
}
