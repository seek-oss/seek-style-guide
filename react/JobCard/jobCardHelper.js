import PropTypes from 'prop-types';
import Constants from '../Constants/Constants';

const { JOBADTYPE_JOBSDB_DEFAULT, JOBADTYPE_JOBSDB_BRANDED, JOBADTYPE_JOBSTREET_DEFAULT, JOBADTYPE_JOBSTREET_STANDOUT } = Constants;

const jobsdbDefault = {
  showCompanyLogo: false,
  showCompanyPic: false,
  showHighlightedBg: false,
  showSellingPoint: true,
  showDescription: false
};
const jobsdbBranded = {
  showCompanyLogo: true,
  showCompanyPic: false,
  showHighlightedBg: false,
  showSellingPoint: true,
  showDescription: false
};
const jobstreetDefault = {
  showCompanyLogo: true,
  showCompanyPic: false,
  showHighlightedBg: false,
  showSellingPoint: false,
  showDescription: true
};
const jobstreetStandout = {
  showCompanyLogo: true,
  showCompanyPic: true,
  showHighlightedBg: true,
  showSellingPoint: true,
  showDescription: true
};

const getJobAdTypeOption = jobAdType => {
  switch (jobAdType) {
    case JOBADTYPE_JOBSDB_DEFAULT:
      return jobsdbDefault;
    case JOBADTYPE_JOBSDB_BRANDED:
      return jobsdbBranded;
    case JOBADTYPE_JOBSTREET_DEFAULT:
      return jobstreetDefault;
    case JOBADTYPE_JOBSTREET_STANDOUT:
      return jobstreetStandout;
    default:
      return {};
  }
};

export default getJobAdTypeOption;
getJobAdTypeOption.propTypes = {
  jobAdType: PropTypes.string
};
