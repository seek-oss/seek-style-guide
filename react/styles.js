import PinkButton from './buttons/PinkButton/PinkButton.less';

export {
  PinkButton
};
