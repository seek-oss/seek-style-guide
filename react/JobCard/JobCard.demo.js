import React from 'react';
import PropTypes from 'prop-types';
import { JobCard, PageBlock, Constants } from 'seek-asia-style-guide/react';

const { JOBADTYPE_JOBSDB_DEFAULT, JOBADTYPE_JOBSDB_BRANDED, JOBADTYPE_JOBSTREET_DEFAULT, JOBADTYPE_JOBSTREET_STANDOUT } = Constants;

const JobCardContainer = ({ component: DemoComponent, componentProps }) => {
  return (
    <PageBlock style={{ width: '100%' }}>
      <DemoComponent {...componentProps} />
    </PageBlock>
  );
};

JobCardContainer.propTypes = {
  component: PropTypes.element,
  componentProps: PropTypes.object
};

export default {
  route: '/jobCard',
  title: 'Job Card',
  component: JobCard,
  container: JobCardContainer,
  initialProps: {
    job: {
      company: 'SEEK Asia',
      jobTitle: 'Senior Software Engineer (6 months Contract)',
      jobUrl: 'https://www-dev.jobstreet.com.my/en/job/20171002-3-senior-front-end-developer-update-x-2-6100835/origin/dev/sources/3?fr=J',
      location: 'Kuala Lumpur, Selangor, Johor Bahru',
      description: 'Responsibilities :Responsible for Client Relationship Management and Worker Performance Management. Responsible for full spectrum of human resource and admin function, include...',
      companyLogoUrl: 'https://siva.jsstatic.com/my/94463/images/logo/94463_logo_0_48885.png',
      companyPictureUrl: 'https://siva.jsstatic.com/my/94463/images/photo/94463_photo_0_621506.jpg',
      postingDuration: '1 hour ago',
      salary: 'RM99999 - RM999999',
      sellingPoints: [
        'We practice a vibrant & energetic office culture',
        'Our company supports a fun yet balanced working environment',
        'We support a safe environment for our employees'
      ]
    },
    jobAdType: JOBADTYPE_JOBSDB_DEFAULT
  },
  options: [
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Featured',
          transformProps: ({ className, ...props }) => ({
            ...props,
            job: {
              ...props.job,
              featuredLabel: 'Featured'
            }
          })
        },
        {
          label: 'Company Confidential',
          transformProps: ({ className, ...props }) => ({
            ...props,
            job: {
              ...props.job,
              company: '',
              confidentialLabel: 'Company Confidential'
            }
          })
        },
        {
          label: 'Classified',
          transformProps: ({ className, ...props }) => ({
            ...props,
            job: {
              ...props.job,
              classifiedLabel: 'Classified'
            }
          })
        },
        {
          label: 'No Salary',
          transformProps: ({ className, ...props }) => ({
            ...props,
            job: {
              ...props.job,
              salary: null
            }
          })
        },
        {
          label: 'Highlight Keyword',
          transformProps: ({ className, ...props }) => ({
            ...props,
            job: {
              ...props.job
            },
            keyword: 'seek senior engineer'
          })
        }
      ]
    },
    {
      label: 'JobAd Type',
      type: 'radio',
      states: [
        {
          label: 'jobsDB standard',
          transformProps: props => ({
            ...props,
            jobAdType: JOBADTYPE_JOBSDB_DEFAULT
          })
        },
        {
          label: 'jobsDB Branded',
          transformProps: props => ({
            ...props,
            jobAdType: JOBADTYPE_JOBSDB_BRANDED
          })
        },
        {
          label: 'jobStreet Default',
          transformProps: props => ({
            ...props,
            jobAdType: JOBADTYPE_JOBSTREET_DEFAULT
          })
        },
        {
          label: 'jobStreet Standout',
          transformProps: props => ({
            ...props,
            jobAdType: JOBADTYPE_JOBSTREET_STANDOUT
          })
        }
      ]
    },
    {
      label: 'Company Pic',
      type: 'radio',
      states: [
        {
          label: 'Normal',
          transformProps: props => ({
            ...props,
            job: {
              ...props.job,
              companyPictureUrl: 'https://siva.jsstatic.com/my/94463/images/photo/94463_photo_0_621506.jpg'
            }
          })
        },
        {
          label: 'Very long',
          transformProps: props => ({
            ...props,
            job: {
              ...props.job,
              companyPictureUrl: 'https://siva.jsstatic.com/my/56932/images/photo/56932_photo_0_213672.jpg'
            }
          })
        },
        {
          label: 'Vertical',
          transformProps: props => ({
            ...props,
            job: {
              ...props.job,
              companyPictureUrl: 'https://siva.jsstatic.com/my/86688/images/photo/86688_photo_0_400935.jpg'
            }
          })
        },
        {
          label: 'Square',
          transformProps: props => ({
            ...props,
            job: {
              ...props.job,
              companyPictureUrl: 'https://siva.jsstatic.com/my/119697/images/photo/119697_photo_0_738574.jpg'
            }
          })
        }
      ]
    }
  ]
};
