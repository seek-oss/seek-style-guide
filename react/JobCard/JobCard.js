import React from 'react';
import PropTypes from 'prop-types';

import Text from '../Text/Text';
import Card from '../Card/Card';
import Section from '../Section/Section';
import LocationIcon from '../LocationIcon/LocationIcon';
import MoneyIcon from '../MoneyIcon/MoneyIcon';
import styles from './JobCard.less';
import classnames from 'classnames';
import match from 'autosuggest-highlight/match';
import parse from 'autosuggest-highlight/parse';
import getJobAdTypeOption from './jobCardHelper.js';

const getParts = (text, query) => {
  if (!text || !query) {
    return null;
  }
  const matches = match(text, query);
  // No point to parse if matches is empty array
  if (matches.length === 0) {
    return null;
  }
  return parse(text, matches);
};

const JobCard = ({ job, keyword = '', jobAdType }) => {
  const jobAdTypeOption = getJobAdTypeOption(jobAdType);
  let title = <Text waving semiStrong className={styles.positionTitle}>{job.jobTitle}</Text>;
  let company = job.company && <span className={styles.companyName}>{job.company}</span> || '';
  const keywordParts = getParts(job.jobTitle, keyword);
  const companyParts = getParts(job.company, keyword);
  if (keywordParts) {
    title = (
      <div>
        {
          keywordParts.map((part, index) => {
            return (
              <Text
                strong={part.highlight}
                semiStrong={!part.highlight}
                waving
                className={classnames(styles.positionTitle, { [styles.titleKeyword]: part.highlight })}
                key={index}>
                {part.text}
              </Text>
            );
          })
        }
      </div>
    );
  }
  if (companyParts) {
    company = (
      <span className={styles.companyName}>
        {
          companyParts.map((part, index) => {
            return (
              <span
                className={part.highlight ? styles.highlight : null}
                key={index}>
                {part.text}
              </span>
            );
          })
        }
      </span>
    );
  }
  return (
    <Card className={classnames(styles.root, { [styles.highlightedBg]: jobAdTypeOption.showHighlightedBg })}>
      <Section className={styles.headerSection}>
        {title}
      </Section>
      <Section className={styles.bodySection}>
        <div>
          <Text intimate className={styles.company}>
            {job.featuredLabel && (<span className={styles.featuredLabel}>{job.featuredLabel}</span>)}
            {job.classifiedLabel && (<span className={styles.classifiedLabel}>{job.classifiedLabel}</span>)}
            {job.confidentialLabel && (<span className={styles.confidentialLabel}>{job.confidentialLabel}</span>)}
            {company}
          </Text>
          {jobAdTypeOption.showSellingPoint && job.sellingPoints && (
            <div
              className={classnames(styles.sellingPointsSection, { [styles.withDescription]: jobAdTypeOption.showDescription && job.description })}>
              <ul className={styles.sellingPointsList} >
                {job.sellingPoints.map((sellingPoint, i) => {
                  return (
                    <li key={i}><Text intimate className={styles.sellingPoint}>{sellingPoint}</Text></li>
                  );
                })}
              </ul>
            </div>
          )}
          {jobAdTypeOption.showDescription && job.description && (
            <div className={styles.jobDescriptionSection}>
              <Text intimate className={styles.bodyDescriptionText}>{job.description}</Text>
            </div>
          )}
        </div>
        {jobAdTypeOption.showCompanyPic && job.companyPictureUrl && (
          <div className={styles.companyPicWrapper}>
            <img className={styles.companyPic} src={job.companyPictureUrl} />
          </div>
        )}
      </Section>
      <Section className={styles.footerSection}>
        <div className={styles.footerLeft}>
          <div className={styles.jobInfoContainer}>
            <div className={styles.jobInfoList}>
              <Text intimate className={styles.jobInfo}><LocationIcon className={styles.jobInfoIcon} /> {job.location}</Text>
              {job.salary && (<Text intimate className={styles.jobInfo}><MoneyIcon className={styles.jobInfoIcon} /> {job.salary}</Text>)}
            </div>
            <Text whispering className={styles.postingDuration}>{job.postingDuration}</Text>
          </div>
        </div>
        {jobAdTypeOption.showCompanyLogo && (
          <div className={styles.companyLogoWrapper}>
            <img className={styles.companyLogo} src={job.companyLogoUrl} />
          </div>
        )}
      </Section>
    </Card>
  );
};

export default JobCard;
JobCard.propTypes = {
  keyword: PropTypes.string,
  job: PropTypes.shape({
    company: PropTypes.string,
    jobTitle: PropTypes.string.isRequired,
    jobUrl: PropTypes.string.isRequired,
    sellingPoints: PropTypes.arrayOf(PropTypes.string),
    companyPictureUrl: PropTypes.string,
    companyLogoUrl: PropTypes.string,
    description: PropTypes.string,
    location: PropTypes.string.isRequired,
    salary: PropTypes.string,
    postingDuration: PropTypes.string.isRequired,
    featuredLabel: PropTypes.string,
    classifiedLabel: PropTypes.string,
    confidentialLabel: PropTypes.string
  }).isRequired,
  jobAdType: PropTypes.string
};
