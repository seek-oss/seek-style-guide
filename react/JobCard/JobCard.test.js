import React from 'react';
import { shallow } from 'enzyme';

import JobCard from './JobCard';
import { Constants } from 'seek-asia-style-guide/react';

const { JOBADTYPE_JOBSDB_DEFAULT, JOBADTYPE_JOBSTREET_STANDOUT } = Constants;

const defaultJob = {
  company: 'SEEK Asia',
  jobTitle: 'Senior Software Engineer (6 months Contract)',
  jobUrl: 'https://www-dev.jobstreet.com.my/en/job/20171002-3-senior-front-end-developer-update-x-2-6100835/origin/dev/sources/3?fr=J',
  location: 'Kuala Lumpur',
  description: 'Responsibilities :Responsible for Client Relationship Management and Worker Performance Management. Responsible for full spectrum of human resource and admin function, include...',
  companyLogoUrl: 'https://siva.jsstatic.com/my/94463/images/logo/94463_logo_0_48885.png',
  postingDuration: '1 hour ago',
  salary: 'RM99999 - RM999999'
};

describe('JobCard', () => {
  it('should render with default props', () => {
    const wrapper = shallow(<JobCard job={defaultJob} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with featured label', () => {
    const descriptionJob = {
      ...defaultJob,
      featuredLabel: 'Feature'
    };

    const wrapper = shallow(<JobCard job={descriptionJob} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with company confidential in grey label', () => {
    const descriptionJob = {
      ...defaultJob,
      confidentialLabel: 'Company Confidential',
      company: null
    };

    const wrapper = shallow(<JobCard job={descriptionJob} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with Classified in grey label', () => {
    const descriptionJob = {
      ...defaultJob,
      classifiedLabel: 'Classified'
    };

    const wrapper = shallow(<JobCard job={descriptionJob} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with selling points props', () => {
    const descriptionJob = {
      ...defaultJob,
      sellingPoints: [
        'We practice a vibrant & energetic office culture',
        'Our company supports a fun yet balanced working environment',
        'We support a safe environment for our employees'
      ]
    };

    const wrapper = shallow(<JobCard job={descriptionJob} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render job title with bold keyword', () => {
    const descriptionJob = {
      ...defaultJob
    };
    const keyword = 'Seek Senior Engineer';
    const wrapper = shallow(<JobCard job={descriptionJob} keyword={keyword} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render company with bold keyword', () => {
    const descriptionJob = {
      ...defaultJob
    };
    const keyword = 'Seek';
    const wrapper = shallow(<JobCard job={descriptionJob} keyword={keyword} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render jobsDB default style', () => {
    const descriptionJob = {
      ...defaultJob
    };
    const keyword = 'Seek';
    const wrapper = shallow(<JobCard job={descriptionJob} keyword={keyword} jobAdType={JOBADTYPE_JOBSDB_DEFAULT} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render jobStreet standout style', () => {
    const descriptionJob = {
      ...defaultJob
    };
    const keyword = 'Seek';
    const wrapper = shallow(<JobCard job={descriptionJob} keyword={keyword} jobAdType={JOBADTYPE_JOBSTREET_STANDOUT} />);
    expect(wrapper).toMatchSnapshot();
  });
});
