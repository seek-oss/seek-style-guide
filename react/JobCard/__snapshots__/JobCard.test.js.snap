// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard should render company with bold keyword 1`] = `
<Card
  className="root"
>
  <Section
    className="headerSection"
  >
    <Text
      baseline={true}
      className="positionTitle"
      semiStrong={true}
      waving={true}
    >
      Senior Software Engineer (6 months Contract)
    </Text>
  </Section>
  <Section
    className="bodySection"
  >
    <div>
      <Text
        baseline={true}
        className="company"
        intimate={true}
      >
        <span
          className="companyName"
        >
          <span
            className="highlight"
            key="0"
          >
            SEEK
          </span>
          <span
            className={null}
            key="1"
          >
             Asia
          </span>
        </span>
      </Text>
    </div>
  </Section>
  <Section
    className="footerSection"
  >
    <div
      className="footerLeft"
    >
      <div
        className="jobInfoContainer"
      >
        <div
          className="jobInfoList"
        >
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <LocationIcon
              className="jobInfoIcon"
            />
             
            Kuala Lumpur
          </Text>
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <MoneyIcon
              className="jobInfoIcon"
            />
             
            RM99999 - RM999999
          </Text>
        </div>
        <Text
          baseline={true}
          className="postingDuration"
          whispering={true}
        >
          1 hour ago
        </Text>
      </div>
    </div>
  </Section>
</Card>
`;

exports[`JobCard should render job title with bold keyword 1`] = `
<Card
  className="root"
>
  <Section
    className="headerSection"
  >
    <div>
      <Text
        baseline={true}
        className="positionTitle titleKeyword"
        key="0"
        semiStrong={false}
        strong={true}
        waving={true}
      >
        Senior
      </Text>
      <Text
        baseline={true}
        className="positionTitle"
        key="1"
        semiStrong={true}
        strong={false}
        waving={true}
      >
         Software 
      </Text>
      <Text
        baseline={true}
        className="positionTitle titleKeyword"
        key="2"
        semiStrong={false}
        strong={true}
        waving={true}
      >
        Engineer
      </Text>
      <Text
        baseline={true}
        className="positionTitle"
        key="3"
        semiStrong={true}
        strong={false}
        waving={true}
      >
         (6 months Contract)
      </Text>
    </div>
  </Section>
  <Section
    className="bodySection"
  >
    <div>
      <Text
        baseline={true}
        className="company"
        intimate={true}
      >
        <span
          className="companyName"
        >
          <span
            className="highlight"
            key="0"
          >
            SEEK
          </span>
          <span
            className={null}
            key="1"
          >
             Asia
          </span>
        </span>
      </Text>
    </div>
  </Section>
  <Section
    className="footerSection"
  >
    <div
      className="footerLeft"
    >
      <div
        className="jobInfoContainer"
      >
        <div
          className="jobInfoList"
        >
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <LocationIcon
              className="jobInfoIcon"
            />
             
            Kuala Lumpur
          </Text>
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <MoneyIcon
              className="jobInfoIcon"
            />
             
            RM99999 - RM999999
          </Text>
        </div>
        <Text
          baseline={true}
          className="postingDuration"
          whispering={true}
        >
          1 hour ago
        </Text>
      </div>
    </div>
  </Section>
</Card>
`;

exports[`JobCard should render jobStreet standout style 1`] = `
<Card
  className="root highlightedBg"
>
  <Section
    className="headerSection"
  >
    <Text
      baseline={true}
      className="positionTitle"
      semiStrong={true}
      waving={true}
    >
      Senior Software Engineer (6 months Contract)
    </Text>
  </Section>
  <Section
    className="bodySection"
  >
    <div>
      <Text
        baseline={true}
        className="company"
        intimate={true}
      >
        <span
          className="companyName"
        >
          <span
            className="highlight"
            key="0"
          >
            SEEK
          </span>
          <span
            className={null}
            key="1"
          >
             Asia
          </span>
        </span>
      </Text>
      <div
        className="jobDescriptionSection"
      >
        <Text
          baseline={true}
          className="bodyDescriptionText"
          intimate={true}
        >
          Responsibilities :Responsible for Client Relationship Management and Worker Performance Management. Responsible for full spectrum of human resource and admin function, include...
        </Text>
      </div>
    </div>
  </Section>
  <Section
    className="footerSection"
  >
    <div
      className="footerLeft"
    >
      <div
        className="jobInfoContainer"
      >
        <div
          className="jobInfoList"
        >
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <LocationIcon
              className="jobInfoIcon"
            />
             
            Kuala Lumpur
          </Text>
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <MoneyIcon
              className="jobInfoIcon"
            />
             
            RM99999 - RM999999
          </Text>
        </div>
        <Text
          baseline={true}
          className="postingDuration"
          whispering={true}
        >
          1 hour ago
        </Text>
      </div>
    </div>
    <div
      className="companyLogoWrapper"
    >
      <img
        className="companyLogo"
        src="https://siva.jsstatic.com/my/94463/images/logo/94463_logo_0_48885.png"
      />
    </div>
  </Section>
</Card>
`;

exports[`JobCard should render jobsDB default style 1`] = `
<Card
  className="root"
>
  <Section
    className="headerSection"
  >
    <Text
      baseline={true}
      className="positionTitle"
      semiStrong={true}
      waving={true}
    >
      Senior Software Engineer (6 months Contract)
    </Text>
  </Section>
  <Section
    className="bodySection"
  >
    <div>
      <Text
        baseline={true}
        className="company"
        intimate={true}
      >
        <span
          className="companyName"
        >
          <span
            className="highlight"
            key="0"
          >
            SEEK
          </span>
          <span
            className={null}
            key="1"
          >
             Asia
          </span>
        </span>
      </Text>
    </div>
  </Section>
  <Section
    className="footerSection"
  >
    <div
      className="footerLeft"
    >
      <div
        className="jobInfoContainer"
      >
        <div
          className="jobInfoList"
        >
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <LocationIcon
              className="jobInfoIcon"
            />
             
            Kuala Lumpur
          </Text>
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <MoneyIcon
              className="jobInfoIcon"
            />
             
            RM99999 - RM999999
          </Text>
        </div>
        <Text
          baseline={true}
          className="postingDuration"
          whispering={true}
        >
          1 hour ago
        </Text>
      </div>
    </div>
  </Section>
</Card>
`;

exports[`JobCard should render with Classified in grey label 1`] = `
<Card
  className="root"
>
  <Section
    className="headerSection"
  >
    <Text
      baseline={true}
      className="positionTitle"
      semiStrong={true}
      waving={true}
    >
      Senior Software Engineer (6 months Contract)
    </Text>
  </Section>
  <Section
    className="bodySection"
  >
    <div>
      <Text
        baseline={true}
        className="company"
        intimate={true}
      >
        <span
          className="classifiedLabel"
        >
          Classified
        </span>
        <span
          className="companyName"
        >
          SEEK Asia
        </span>
      </Text>
    </div>
  </Section>
  <Section
    className="footerSection"
  >
    <div
      className="footerLeft"
    >
      <div
        className="jobInfoContainer"
      >
        <div
          className="jobInfoList"
        >
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <LocationIcon
              className="jobInfoIcon"
            />
             
            Kuala Lumpur
          </Text>
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <MoneyIcon
              className="jobInfoIcon"
            />
             
            RM99999 - RM999999
          </Text>
        </div>
        <Text
          baseline={true}
          className="postingDuration"
          whispering={true}
        >
          1 hour ago
        </Text>
      </div>
    </div>
  </Section>
</Card>
`;

exports[`JobCard should render with company confidential in grey label 1`] = `
<Card
  className="root"
>
  <Section
    className="headerSection"
  >
    <Text
      baseline={true}
      className="positionTitle"
      semiStrong={true}
      waving={true}
    >
      Senior Software Engineer (6 months Contract)
    </Text>
  </Section>
  <Section
    className="bodySection"
  >
    <div>
      <Text
        baseline={true}
        className="company"
        intimate={true}
      >
        <span
          className="confidentialLabel"
        >
          Company Confidential
        </span>
      </Text>
    </div>
  </Section>
  <Section
    className="footerSection"
  >
    <div
      className="footerLeft"
    >
      <div
        className="jobInfoContainer"
      >
        <div
          className="jobInfoList"
        >
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <LocationIcon
              className="jobInfoIcon"
            />
             
            Kuala Lumpur
          </Text>
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <MoneyIcon
              className="jobInfoIcon"
            />
             
            RM99999 - RM999999
          </Text>
        </div>
        <Text
          baseline={true}
          className="postingDuration"
          whispering={true}
        >
          1 hour ago
        </Text>
      </div>
    </div>
  </Section>
</Card>
`;

exports[`JobCard should render with default props 1`] = `
<Card
  className="root"
>
  <Section
    className="headerSection"
  >
    <Text
      baseline={true}
      className="positionTitle"
      semiStrong={true}
      waving={true}
    >
      Senior Software Engineer (6 months Contract)
    </Text>
  </Section>
  <Section
    className="bodySection"
  >
    <div>
      <Text
        baseline={true}
        className="company"
        intimate={true}
      >
        <span
          className="companyName"
        >
          SEEK Asia
        </span>
      </Text>
    </div>
  </Section>
  <Section
    className="footerSection"
  >
    <div
      className="footerLeft"
    >
      <div
        className="jobInfoContainer"
      >
        <div
          className="jobInfoList"
        >
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <LocationIcon
              className="jobInfoIcon"
            />
             
            Kuala Lumpur
          </Text>
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <MoneyIcon
              className="jobInfoIcon"
            />
             
            RM99999 - RM999999
          </Text>
        </div>
        <Text
          baseline={true}
          className="postingDuration"
          whispering={true}
        >
          1 hour ago
        </Text>
      </div>
    </div>
  </Section>
</Card>
`;

exports[`JobCard should render with featured label 1`] = `
<Card
  className="root"
>
  <Section
    className="headerSection"
  >
    <Text
      baseline={true}
      className="positionTitle"
      semiStrong={true}
      waving={true}
    >
      Senior Software Engineer (6 months Contract)
    </Text>
  </Section>
  <Section
    className="bodySection"
  >
    <div>
      <Text
        baseline={true}
        className="company"
        intimate={true}
      >
        <span
          className="featuredLabel"
        >
          Feature
        </span>
        <span
          className="companyName"
        >
          SEEK Asia
        </span>
      </Text>
    </div>
  </Section>
  <Section
    className="footerSection"
  >
    <div
      className="footerLeft"
    >
      <div
        className="jobInfoContainer"
      >
        <div
          className="jobInfoList"
        >
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <LocationIcon
              className="jobInfoIcon"
            />
             
            Kuala Lumpur
          </Text>
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <MoneyIcon
              className="jobInfoIcon"
            />
             
            RM99999 - RM999999
          </Text>
        </div>
        <Text
          baseline={true}
          className="postingDuration"
          whispering={true}
        >
          1 hour ago
        </Text>
      </div>
    </div>
  </Section>
</Card>
`;

exports[`JobCard should render with selling points props 1`] = `
<Card
  className="root"
>
  <Section
    className="headerSection"
  >
    <Text
      baseline={true}
      className="positionTitle"
      semiStrong={true}
      waving={true}
    >
      Senior Software Engineer (6 months Contract)
    </Text>
  </Section>
  <Section
    className="bodySection"
  >
    <div>
      <Text
        baseline={true}
        className="company"
        intimate={true}
      >
        <span
          className="companyName"
        >
          SEEK Asia
        </span>
      </Text>
    </div>
  </Section>
  <Section
    className="footerSection"
  >
    <div
      className="footerLeft"
    >
      <div
        className="jobInfoContainer"
      >
        <div
          className="jobInfoList"
        >
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <LocationIcon
              className="jobInfoIcon"
            />
             
            Kuala Lumpur
          </Text>
          <Text
            baseline={true}
            className="jobInfo"
            intimate={true}
          >
            <MoneyIcon
              className="jobInfoIcon"
            />
             
            RM99999 - RM999999
          </Text>
        </div>
        <Text
          baseline={true}
          className="postingDuration"
          whispering={true}
        >
          1 hour ago
        </Text>
      </div>
    </div>
  </Section>
</Card>
`;
